src/FffGcodeWriter.cpp

```diff
@@ -828,19 +828,47 @@ void FffGcodeWriter::addMeshLayerToGCode(SliceDataStorage& storage, SliceMeshSto
     }
 }
 
+static void parseIntegerList(const char *input, std::vector<int> *output) {
+    for(;;)
+    {
+        char *endp;
+        int val = strtol(input, &endp, 0);
+        if (endp != input)
+        {
+            output->push_back(val);
+            input = endp;
+            while (*input && isspace(*input))
+                ++input;
+            if (*input == ',')
+                ++input;
+        }
+        else
+            break;
```

Comment:
Rather than have an infinite loop and break it, it is rather simple to pull the conditional out of here and put it as the stopping condition of the for loop. Especially if we're going to use `std::string`.